crates/table/src/eq.rs

```diff
@@ -30,14 +30,23 @@ pub unsafe fn eq_row_in_page(
     fixed_offset_a: PageOffset,
     fixed_offset_b: PageOffset,
     ty: &RowTypeLayout,
+    only_fixed_parts: bool,
 ) -> bool {
+    // Contexts for rows `a` and `b`.
+    let a = BytesPage::new(page_a, fixed_offset_a, ty);
+    let b = BytesPage::new(page_b, fixed_offset_b, ty);
+
+    // If there are only fixed parts in the layout,
+    // there are no pointers to anywhere,
+    // So it is sound to simply check for byte-wise equality
+    // and we need not do the tree traversal at all.
```

Comment:
I would like to see additional justification here that we don't have to worry about padding bytes within the rows causing false negatives. I believe the code is correct, since we zero-initialize pages and never write to padding bytes, but I think that's non-obvious and important to record. E.g. one could imagine a version of `Page` where padding bytes held arbitrary non-poison values, and therefore this comparison could return false for two equal rows.

Reply:
Edit: I actually don't think this is correct in the presence of sums with variants that have distinct layouts.

Imagine a row `Product { x: Sum { full: u64, padding: Product { a: u8, b: u32 } } }`.

1. Insert two rows:
  - A_0: `{ x: full(-1i64 as u64) }`.
  - B_0: `{ x: full(u64::ALTERNATING_BIT_PATTERN) }`.
2. Delete both rows.
3. Insert two new rows, which will occupy the same slots as the previous two rows:
  - A_1: `{ x: padding({ a: 0, b: 0 }) }`.
  - B_1: `{ x: padding({ a: 0, b: 0 }) }`.
4. Compare the two new rows with `eq_row_in_page`. Expect `true`, as they are equal, but get `false`, since `A_0[1..4] != B_0[1..4]`.

Reply:
I'm a little surprised too. I don't actually care, though; I think these new numbers seem like a reasonable and productive trade-off.